docs/useful-snippets/snippets.rst

```diff
@@ -58,3 +58,57 @@ code as follows.
 If you would still like to run your code in Jupyter you will need to adjust the cell as noted above. Then you can
 you can add the `nbmultitask library <(https://nbviewer.org/github/micahscopes/nbmultitask/blob/39b6f31b047e8a51a0fcb5c93ae4572684f877ce/examples.ipynb)>`__
 or look at this `stackoverflow <https://stackoverflow.com/questions/50937362/multiprocessing-on-python-3-jupyter>`__.
+
+Using multiple schedules and data collectors
+-------------
+
+One may occasionally want multiple schedules with distinct sets of agents and collect data on each of those schedules independently while allowing the agents to interact in the same model. This would be particularly useful if the agents in your model do not all have the same attributes and you want to collect data on them in groups. You can create a simple scheduler class to mix the agents in each schedule and data collect on each schedule independently.
+
+.. code:: python
+
+  from random import shuffle
+
+  # simple schedule mixer
+  class ScheduleMixerRandom:
+      """
+      Be careful: ScheduleMixerRandom as written here would not capture changes to
+      the agent lists in the sub-schedules. The user would need to be mindful if
+      this functionality were to be included.
+      """
+
+      def __init__(self, schedules):
+          self._agents = {}
+          for s in schedules:
+          for a in s.agents:
+              self._agents[a.unique_id] = a # each agent must have a unique id across models
+
+      def step(self):
+          # do stuff with self._agents
+          ...
+
+  class MyModel(Model):
+      def __init__(self, ...):
+          super().__init__()
+          self.schedule_one = BaseScheduler()
+          self.schedule_two = BaseScheduler()
+
+          # each DataCollector instance is associated with a specific schedule (and therefore its agents)
+          self.datacollector_one = DataCollector(agent_reporters = {'value_one': 'value_one'}, schedule = self.schedule_one)
+          self.datacollector_two = DataCollector(agent_reporters = {'value_two': 'value_two'}, schedule = self.schedule_two)
+
+          self.schedule_mixer = ScheduleMixerRandom(schedules=[self.schedule_one, self.schedule_two])
+
+      def step():
+          self.schedule_mixer.step() # calls step() on each agent in both schedules, mixed together
```

Comment:
To save extra review roundtrip: the comments should be in their own lines, so that the line doesn't exceed 88 chars length limit imposed by Black.

Reply:
No problem.